client/packages/common/src/intl/locales/index.ts

```diff
@@ -18,3 +18,5 @@ type WithOrWithoutPlural<K> = TypeOptions['jsonFormat'] extends 'v4'
   : K;
 
 export type LocaleKey = WithOrWithoutPlural<keyof typeof common>;
+
+export const localeKeySet = new Set(Object.keys(common));
```

Comment:
This isn't strictly necessary as we can cast a string 'as LocaleKey', but I thought it would be nice to be able to check for type safety without casting.